app/main.py

```diff
@@ -1,214 +1,27 @@
-        "power": 35,
-        "hp": 100,
-        "armour": [],
-        "weapon": {
-            "name": "Metal Sword",
-            "power": 50,
-        },
-        "potion": None,
-    },
-    "arthur": {
-        "name": "Artur",
-        "power": 45,
-        "hp": 75,
-        "armour": [
-            {
-                "part": "helmet",
-                "protection": 15,
-            },
-            {
-                "part": "breastplate",
-                "protection": 20,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Two-handed Sword",
-            "power": 55,
-        },
-        "potion": None,
-    },
-    "mordred": {
-        "name": "Mordred",
-        "power": 30,
-        "hp": 90,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 15,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Poisoned Sword",
-            "power": 60,
-        },
-        "potion": {
-            "name": "Berserk",
-            "effect": {
-                "power": +15,
-                "hp": -5,
-                "protection": +10,
-            }
-        }
-    },
-    "red_knight": {
-        "name": "Red Knight",
-        "power": 40,
-        "hp": 70,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 25,
-            }
-        ],
-        "weapon": {
-            "name": "Sword",
-            "power": 45
-        },
-        "potion": {
-            "name": "Blessing",
-            "effect": {
-                "hp": +10,
-                "power": +5,
-            }
-        }
-    }
-}
+from typing import Dict
 
+from app.warriors import Knight
+from app.data import KNIGHTS
 
-def battle(knightsConfig):
-    # BATTLE PREPARATIONS:
 
-    # lancelot
-    lancelot = knightsConfig["lancelot"]
+def battle(knights_config: Dict[str, dict]) -> Dict[str, int]:
+    knights = {}
 
-    # apply armour
-    lancelot["protection"] = 0
-    for a in lancelot["armour"]:
-        lancelot["protection"] += a["protection"]
+    for warrior, config in knights_config.items():
+        knight = Knight(name=config["name"],
+                        power=config["power"],
+                        hp=config["hp"],
+                        weapon=config["weapon"],
+                        armour=config["armour"],
+                        potion=config["potion"])
+        knights[warrior] = knight
 
-    # apply weapon
-    lancelot["power"] += lancelot["weapon"]["power"]
+    knights["lancelot"].strike_enemy(knights["mordred"])
+    knights["mordred"].strike_enemy(knights["lancelot"])
+    knights["arthur"].strike_enemy(knights["red_knight"])
+    knights["red_knight"].strike_enemy(knights["arthur"])
```

Comment:
try to rewrite this method, so there won't need to call it for every knight
U can make this method work for pair